doc/src/tutorials/biomechanics/biomechanics.rst

```diff
@@ -0,0 +1,1376 @@
+   >>> Fr, Frs = kane.kanes_equations((block,), (muscle_loads + [gravity]))
+
+The equations of motion are made up of the kinematical differential equation,
+the dynamical differential equation (Newton's Second Law), and the muscle
+activation differential equation. The explicit form of each can be formed like
+so:
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> dqdt = u
+   >>> dudt = kane.forcing[0]/m
+   >>> dadt = muscle.rhs()[0]
+
+We can now create a numerical function that evaluates the equations of motion
+given the state, inputs, and constant parameters. Start by listing each
+symbolically:
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> a = muscle.a
+   >>> e = muscle.e
+   >>> state = [q, u, a]
+   >>> inputs = [e]
+   >>> constants = [m, g, F_M_max, l_M_opt, l_T_slack, v_M_max, alpha_opt, beta]
+
+Then the numerical function to evaluate the right hand side of the explicit
+ordinary differential equations is:
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> eval_eom = sm.lambdify((state, inputs, constants), (dqdt, dudt, dadt))
+
+It will additionally be interesting to numerically evaluate the muscle force,
+so create a function for it too:
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> force = muscle.force.xreplace({q.diff(): u})
+   >>> eval_force = sm.lambdify((state, constants), force)
+
+To test these functions we need some suitable numerical values. This muscle
+will be able to produce a maximum force of 10 N to lift a mass of 0.5 kg:
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> import numpy as np
+   >>> p_vals = np.array([
+   ...     0.5,  # m [kg]
+   ...     9.81,  # g [m/s/s]
+   ...     10.0,  # F_M_max [N]
+   ...     0.18,  # l_M_opt [m]
+   ...     0.17,  # l_T_slack [m]
+   ...     10.0,  # v_M_max [m/s]
+   ...     0.0,  # alpha_opt
+   ...     0.1,  # beta
+   ... ])
+   ...
+
+Our tendon is rigid, so the length of the muscle will be
+:math:`q-l_{T_\textrm{slack}}` and we want to give an initial muscle length
+near its force producing peak, so we choose :math:`q_0=l_{M_\textrm{opt}} +
+l_{T_\textrm{slack}}`:
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> x_vals = np.array([
+   ...     p_vals[3] + p_vals[4],  # q [m]
+   ...     0.0,  # u [m/s]
+   ...     0.0,  # a [unitless]
+   ... ])
+   ...
+
+Set the excitation to 1.0 and test the numerical functions:
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> r_vals = np.array([
+   ...     1.0,  # e
+   ... ])
+   ...
+   >>> eval_eom(x_vals, r_vals, p_vals)
+   (0.0, 9.81, 133.33333307568913)
+   >>> eval_force(x_vals, p_vals)
+   1.4499681738213515e-16
```

Comment:
This is passing for me locally with both `pytest --doctest-only` and `bin/doctest`. SymPy is still using it's own doctest runner behind the scenes, which doesn't have a `FLOAT_CMP` setting like pytest with doctestplus. Not sure what the best solution here is. Whether there's a neat solution, if we have to change the value of `x_vals` so that the force is nonzero and then we can use `# doctest: +ELLIPSES`, or if we just use `# doctest: +SKIP`.

Reply:
We could just print the first few sig figs.

```
print('{:0.3f}'.format(eval_force(x_vals, p_vals)))
```

Or skip the doctest.